modules/widgets/internet-defense-league.php

```diff
@@ -0,0 +1,154 @@
+<?php /*
+
+**************************************************************************
+
+Plugin Name:  Internet Defense League
+Description:  Displays your support for the Internet Defense League.
+Author:       Automattic Inc.
+Author URI:   https://automattic.com/
+
+**************************************************************************/
+
+class Internet_Defense_League_Widget extends WP_Widget {
+
+	public $defaults = array();
+
+	public $variant;
+	public $variants = array();
+
+	public $campaign;
+	public $campaigns  = array();
+	public $no_current = true;
+
+	public $badge;
+	public $badges = array();
+
+	function __construct() {
+		parent::__construct(
+			'internet_defense_league_widget',
+			apply_filters( 'jetpack_widget_name_widget', esc_html__( 'Internet Defense League', 'jetpack' ) ),
+			array(
+				'description' => __( 'Show your support for the Internet Defense League.', 'internetdefenseleague' ),
+			)
+		);
+
+		// When enabling campaigns other than 'none' or empty, change $no_current to false above.
+		$this->campaigns = array(
+			''       => __( 'All current and future campaigns', 'internetdefenseleague' ),
+			// 'nsa' => __( 'NSA Protest on July 4th, 2013', 'internetdefenseleague' ),
+			'none'   => __( 'None, just display the badge please', 'internetdefenseleague' ),
+		);
+
+		$this->variants = array(
+			'banner' => __( 'Banner at the top of my site', 'internetdefenseleague' ),
+			'modal'  => __( 'Modal (Overlay Box)', 'internetdefenseleague' ),
+		);
+
+		$this->badges = array(
+			'shield_badge'   => __( 'Shield Badge', 'internetdefenseleague' ),
+			'super_badge'    => __( 'Super Badge', 'internetdefenseleague' ),
+			'side_bar_badge' => __( 'Red Cat Badge', 'internetdefenseleague' ),
+			'none'           => __( 'Don\'t display a badge (just the campaign)', 'internetdefenseleague' ),
+		);
+
+		$this->defaults = array(
+			'campaign' => key( $this->campaigns ),
+			'variant'  => key( $this->variants ),
+			'badge'    => key( $this->badges ),
+		);
+	}
+
+	public function widget( $args, $instance ) {
+		$instance = wp_parse_args( $instance, $this->defaults );
+
+		if ( 'none' != $instance['badge'] ) {
+			if ( ! isset( $this->badges[ $instance['badge'] ] ) ) {
+				$instance['badge'] = $this->defaults['badge'];
+			}
+			echo $args['before_widget'];
+			echo '<p><a href="https://internetdefenseleague.org/"><img src="' . esc_url( 'https://internetdefenseleague.org/images/badges/final/' . $instance['badge'] . '.png' ) . '" alt="Member of The Internet Defense League" style="max-width: 100%; height: auto;" /></a></p>';
```

Comment:
Updated in 70ca32a

![screen shot 2017-04-21 at 3 54 26 pm](https://cloud.githubusercontent.com/assets/1270189/25298580/fa9ebe5c-26aa-11e7-96a5-f0c69df06793.png)

Should I also add translation strings here for the alt text?